internal/controller/storageclient_controller.go

```diff
@@ -779,27 +661,33 @@ func removeStorageClaimAsOwner(obj client.Object) {
 	}
 }
 
-func (r *storageClientReconcile) EnsureClusterScopedResource(obj client.Object, extRes *providerpb.ExternalResource, useReplace bool) error {
-	obj.SetName(extRes.Name)
+func (r *storageClientReconcile) ensureResource(obj client.Object, rawObject []byte, useReplace bool, clusterScoped bool) error {
+	objectMeta := metav1.ObjectMeta{}
+	if err := json.Unmarshal(rawObject, &objectMeta); err != nil {
+		return err
+	}
+
+	obj.SetName(objectMeta.Name)
+	if !clusterScoped {
+		obj.SetNamespace(r.OperatorNamespace)
+	}
 
 	mutateFunc := func() error {
-		if err := json.Unmarshal(extRes.Data, obj); err != nil {
+		if err := json.Unmarshal(rawObject, obj); err != nil {
 			return fmt.Errorf("failed to unmarshal %s configuration response: %v", obj.GetName(), err)
 		}
```

Comment:
Is the intent to completely ignore any local metadata and just override everything?
Also what will happen to resources with a Status sub struct that is not a pointer?